samples/apply-stretch-renderer/src/main/java/com/esri/arcgismaps/sample/applystretchrenderer/components/ApplyStretchRendererViewModel.kt

```diff
@@ -0,0 +1,260 @@
+    val maxValue = _maxValue.asStateFlow()
+
+    // Percent clip parameters (values represent percent 0..100)
+    private val _percentMin = MutableStateFlow(DEFAULT_PERCENT_MIN)
+    val percentMin = _percentMin.asStateFlow()
+
+    private val _percentMax = MutableStateFlow(DEFAULT_PERCENT_MAX)
+    val percentMax = _percentMax.asStateFlow()
+
+    // Standard deviation factor (typical range 0.25..4.0)
+    private val _stdDeviationFactor = MutableStateFlow(DEFAULT_STD_DEVIATION_FACTOR)
+    val stdDeviationFactor = _stdDeviationFactor.asStateFlow()
+
+    init {
+        viewModelScope.launch {
+            rasterLayer.load().onSuccess {
+                arcGISMap.operationalLayers.add(rasterLayer)
+            }.onFailure {
+                messageDialogVM.showMessageDialog(it)
+            }
+
+            arcGISMap.load().onFailure {
+                messageDialogVM.showMessageDialog(it)
+            }
+
+            val center = rasterLayer.fullExtent?.center ?: Point(0.0, 0.0, SpatialReference.wgs84())
+            mapViewProxy.setViewpoint(Viewpoint(center = center, scale = 80_000.0))
+            updateRenderer()
+        }
+    }
+
+    /** Update the current stretch type by index from [stretchTypeOptions]. */
+    fun updateStretchTypeByIndex(index: Int) {
+        val safeIndex = index.coerceIn(0, stretchTypeOptions.lastIndex)
+        _selectedStretchType.value = when (safeIndex) {
+            0 -> StretchType.MinMax
+            1 -> StretchType.PercentClip
+            else -> StretchType.StandardDeviation
+        }
+    }
+
+    /** Update MinMax minimum value (clamped to 0..(max-1)). */
+    fun updateMinValue(value: Double) {
+        val max = _maxValue.value
+        _minValue.value = value.coerceIn(DEFAULT_MIN, max - 1.0)
+    }
+
+    /** Update MinMax maximum value (clamped to (min+1)..255). */
+    fun updateMaxValue(value: Double) {
+        val min = _minValue.value
+        _maxValue.value = value.coerceIn(min + 1.0, 255.0)
+    }
+
+    /** Update Percent Clip minimum percent (clamped to 0..percentMax). */
+    fun updatePercentMin(value: Double) {
+        val max = _percentMax.value
+        _percentMin.value = value.coerceIn(DEFAULT_MIN, max)
+    }
+
+    /** Update Percent Clip maximum percent (clamped to percentMin..100). */
+    fun updatePercentMax(value: Double) {
+        val min = _percentMin.value
+        _percentMax.value = value.coerceIn(min, 100.0)
+    }
+
+    /** Update Standard Deviation factor (clamped to 0.25..4.0). */
+    fun updateStdDeviationFactor(value: Double) {
+        _stdDeviationFactor.value = value.coerceIn(0.25, 4.0)
+    }
+
+    /** Construct and apply a StretchRenderer to the raster layer using current UI parameters. */
+    fun updateRenderer() {
+        appliedStretchType = _selectedStretchType.value
+        val parameters: StretchParameters = when (_selectedStretchType.value) {
+            StretchType.MinMax -> {
+                // save the new initial values
```

Comment:
Nitpick suggestion, to use capitalization in the in line comments here, (and below in `updateRenderer`)
```suggestion
                // Save the new initial values
```